StenosisMeasurement2D/StenosisMeasurement2D.py

```diff
@@ -0,0 +1,750 @@
+from slicer.ScriptedLoadableModule import *
+from slicer.util import VTKObservationMixin
+
+#
+# StenosisMeasurement2D
+#
+
+class StenosisMeasurement2D(ScriptedLoadableModule):
+  """Uses ScriptedLoadableModule base class, available at:
+  https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+  """
+
+  def __init__(self, parent):
+    ScriptedLoadableModule.__init__(self, parent)
+    self.parent.title = "Stenosis measurement : 2D"
+    self.parent.categories = ["Vascular Modeling Toolkit"]
+    self.parent.dependencies = []
+    self.parent.contributors = ["SET [Surgeon] [Hobbyist developer]", "Andras Lasso, PerkLab"]
+    self.parent.helpText = """
+This <a href="https://github.com/vmtk/SlicerExtension-VMTK/">module</a> calculates the surface area of segments cut by a slice plane in its orientation. It is intended for quick two dimensional arterial stenosis evaluation, but is actually purpose agnostic.
+"""
+    # TODO: replace with organization, grant and thanks
+    self.parent.acknowledgementText = """
+This file was originally developed by Jean-Christophe Fillion-Robin, Kitware Inc., Andras Lasso, PerkLab,
+and Steve Pieper, Isomics, Inc. and was partially funded by NIH grant 3P41RR013218-12S1.
+"""
+
+class StenosisMeasurement2DWidget(ScriptedLoadableModuleWidget, VTKObservationMixin):
+  """Uses ScriptedLoadableModuleWidget base class, available at:
+  https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+  """
+
+  def __init__(self, parent=None):
+    """
+    Called when the user opens the module the first time and the widget is initialized.
+    """
+    ScriptedLoadableModuleWidget.__init__(self, parent)
+    VTKObservationMixin.__init__(self)  # needed for parameter node observation
+    self.logic = None
+    self._parameterNode = None
+    self._updatingGUIFromParameterNode = False
+    """
+    N.B. - We store slice orientation in NthControlPointOrientationMatrixWorld.
+    Unfortunately, this data is not saved in the markups JSON file,
+    though there is an orientation key.
+    """
+    self.currentFiducialNode = None
+    self.currentControlPointID = "-1" # API gets/sets it as a string
+    self.optionResetControlPointOrientationMatrix = False # Checkable menu item
+    self.optionCreateCutModel = False # Checkable menu item
+    """
+    We are using JumpToPointEvent of the fiducial display node.
+    It gets fired 3 times when we click on a control point.
+    We use a timer to run a slot once only.
+    https://discourse.slicer.org/t/observing-a-change-in-sliceoffset-within-a-segmenteditoreffect-class/950/5
+    """
+    # # The slot used by timer cannot have (caller, event) parameters.
+    self.fiducialDisplayNode = None # caller
+    self.fiducialDisplayNodeEvent = None # event
+    self.fiducialDisplayNodeObservation = None
+    self.timer = qt.QTimer()
+    # Menus
+    self.tableMenu = qt.QMenu()
+    self.optionsMenu = qt.QMenu()
```

Comment:
Is this menu shown when you click on a markup point in a view? For that the clean way would be to register custom actions to the ViewContext menu in a subject hierarchy plugin. See example here: https://slicer.readthedocs.io/en/latest/developer_guide/script_repository.html#subject-hierarchy-plugin-offering-view-context-menu-action

Reply:
> Is this menu shown when you click on a markup point in a view?

No, the menu is shown on right click in the 'Options' collapsible area.
![menu_context](https://user-images.githubusercontent.com/33979252/149653683-8fde94aa-b008-41ab-be26-ae84b78e5c53.png)

Reply:
How would users discover that they can bring up a menu by right-clicking there? Why not adding these in a collapsible group ox instead (collapsed by default)? The groupbox would give the hint to the users that they can find additional options there.

Reply:
> How would users discover that they can bring up a menu by right-clicking there?

A tool tip showing "Right click for more" hints for more options. I agree that it's not the *traditional* way in Slicer, I just preferred a lightweight UI. It can of course be changed, please try it once for a personal 'feel' when you have time. I'll change it if you prefer to keep the traditional design.

Reply:
Discoverability and consistency is important. Hiding an instruction in a tooltip to bring up a right-click menu is there does not fulfill any of these. 

If you find that the collapsible groupbox is too big then you can hide the additional buttons using the double-arrow button. That is very small, so it won't make the GUI look complex, but it is still recognizable and used elsewhere in Slicer for this.

Reply:
The option menu is now removed. All options are managed with usual widgets.